google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/RpcMethodMapping.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.conformance.retry;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.Objects.requireNonNull;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.conformance.retry.CtxFunctions.ResourceSetup;
+import com.google.cloud.storage.conformance.retry.Functions.CtxFunction;
+import com.google.common.base.Preconditions;
+import com.google.errorprone.annotations.Immutable;
+import java.util.HashSet;
+import java.util.function.Predicate;
+import org.junit.AssumptionViolatedException;
+
+@Immutable
+final class RpcMethodMapping {
+
+  private final int mappingId;
+  private final RpcMethod method;
+  private final Predicate<TestRetryConformance> applicable;
+  private final CtxFunction setup;
+  private final CtxFunction test;
+  private final CtxFunction tearDown;
+
+  RpcMethodMapping(
+      int mappingId,
+      RpcMethod method,
+      Predicate<TestRetryConformance> applicable,
+      CtxFunction setup,
+      CtxFunction test,
+      CtxFunction tearDown) {
+    this.mappingId = mappingId;
+    this.method = method;
+    this.applicable = applicable;
+    this.setup = setup;
+    this.test = test;
+    this.tearDown = tearDown;
+  }
+
+  public int getMappingId() {
+    return mappingId;
+  }
+
+  public RpcMethod getMethod() {
+    return method;
+  }
+
+  public Predicate<TestRetryConformance> getApplicable() {
+    return applicable;
+  }
+
+  public CtxFunction getSetup() {
+    return setup;
+  }
+
+  public CtxFunction getTest() {
+    return (ctx, c) -> {
+      if (c.isExpectSuccess()) {
+        return test.apply(ctx, c);
+      } else {
+        try {
+          test.apply(ctx, c);
+          fail("expected failure, but succeeded");
+        } catch (StorageException e) {
+          boolean matchExpectedCode = false;
+          int code = e.getCode();
+          HashSet<String> instructions = newHashSet(c.getInstruction().getInstructionsList());
+          if (instructions.contains("return-503") && code == 503) {
```

Comment:
This seems like it needs a TODO for when you have the full set of test cases from the JSON.